app/services/authentication/authenticator.rb

```diff
@@ -113,13 +118,6 @@ def update_profile_updated_at(user)
       user.profile_updated_at = Time.current if user.public_send(field_name)
     end
 
-    def save_identity(identity, user)
```

Comment:
Pulled this all into the transaction...I'm assuming it's okay/safe to do `record_identity_creation` from within the transaction but let me know if I've got that wrong!

Reply:
Technically no. If DataDog raises a HTTP exception the whole transaction will be rolled back, I don't think we want to block a user sign up/login if DD is down or if for any reason the connection times out. `flag_spam_user` which calls Slack asynchronously can stay there as it's correct in the flow. 

See the section "Exception handling and rolling back" in https://api.rubyonrails.org/v5.2/classes/ActiveRecord/Transactions/ClassMethods.html 

You can find an example on error handling within transaction in 

https://github.com/thepracticaldev/dev.to/blob/e0b932e569bb5bcf926f0973032761897834416d/app/services/pro_memberships/biller.rb#L32

Obviously we want to log to Datadog so I think we should have 2 DD calls: one after the transaction which will trigger if everything went well and, optionally, one in an exception handler which triggers if we get an error.

Maybe something like:

```ruby
DatadogStatsClient.increment("identity.errors", tags: ["error:#{err.class}"], message: err.message)
```

cc @mstruve for the best way to log these inline errors